scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -442,22 +442,31 @@ class FossId internal constructor(
         projectCode: String,
         projectName: String
     ): String {
+        val urlWithoutCredentials = if ("@" in url) {
+            log.warn { "The URL should not contain credentials as its interaction with delta scans is unpredictable." }
+
+            url.replaceCredentialsInUri()
+        } else url
+
         // we ignore the revision because we want to do a delta scan
-        val recentScans = scans.recentScansForRepository(url)
+        val recentScans = scans.recentScansForRepository(urlWithoutCredentials)
 
         log.info { "Found ${recentScans.size} scans." }
 
         val existingScan = recentScans.findLatestPendingOrFinishedScan()
 
         val scanCode = if (existingScan == null) {
-            log.info { "No scan found for $url and revision $revision. Creating origin scan..." }
+            log.info { "No scan found for $urlWithoutCredentials and revision $revision. Creating origin scan..." }
             namingProvider.createScanCode(projectName, DeltaTag.ORIGIN)
         } else {
-            log.info { "Scan found for $url and revision $revision. Creating delta scan..." }
+            log.info { "Scan found for $urlWithoutCredentials and revision $revision. Creating delta scan..." }
             namingProvider.createScanCode(projectName, DeltaTag.DELTA)
         }
 
-        val newUrl = if (config.addAuthenticationToUrl) queryAuthenticator(url) else url
+        val newUrl = if (config.addAuthenticationToUrl) {
+            queryAuthenticator(urlWithoutCredentials)
+        } else urlWithoutCredentials
```

Comment:
Formatting: `if` /  `else` should either be both in a single line, or both be blocks. I.e use

```
} else {
    urlWithoutCredentials
}
```